presto-main-base/src/test/java/com/facebook/presto/operator/scalar/TestArraySortByKeyFunctions.java

```diff
@@ -0,0 +1,392 @@
+
+        assertFunction(
+                "array_sort(ARRAY[5, 20, 3, 9, 100], x -> x * CAST(1000000000 AS BIGINT))",
+                new ArrayType(INTEGER),
+                asList(3, 5, 9, 20, 100));
+
+        assertFunction(
+                "array_sort(ARRAY['5', '20', '3', '9', '100'], x -> cast(x as integer))",
+                new ArrayType(createVarcharType(3)),
+                asList("3", "5", "9", "20", "100"));
+
+        assertFunction(
+                "array_sort(ARRAY['5000000000', '20000000000', '3000000000', '9000000000', '100000000000'], x -> cast(x as bigint))",
+                new ArrayType(createVarcharType(12)),
+                asList("3000000000", "5000000000", "9000000000", "20000000000", "100000000000"));
+
+        // Test array_sort_desc
+        assertFunction(
+                "array_sort_desc(ARRAY[5, 20, 3, 9, 100], x -> x + CAST(0.5 AS DOUBLE))",
+                new ArrayType(INTEGER),
+                asList(100, 20, 9, 5, 3));
+
+        assertFunction(
+                "array_sort_desc(ARRAY[5, 20, 3, 9, 100], x -> x * CAST(1000000000 AS BIGINT))",
+                new ArrayType(INTEGER),
+                asList(100, 20, 9, 5, 3));
+
+        assertFunction(
+                "array_sort_desc(ARRAY['5', '20', '3', '9', '100'], x -> cast(x as integer))",
+                new ArrayType(createVarcharType(3)),
+                asList("100", "20", "9", "5", "3"));
+
+        assertFunction(
+                "array_sort_desc(ARRAY['5000000000', '20000000000', '3000000000', '9000000000', '100000000000'], x -> cast(x as bigint))",
+                new ArrayType(createVarcharType(12)),
+                asList("100000000000", "20000000000", "9000000000", "5000000000", "3000000000"));
+    }
+
+    @Test
+    public void testNativeExecutionWithSidecarDisabled()
+    {
+        // Create a test instance with native execution enabled and sidecar explicitly disabled
+        TestArraySortByKeyFunctionsNative nativeTest = new TestArraySortByKeyFunctionsNative();
+        // Manually initialize the function assertions
+        nativeTest.initTestFunctions();
+        try {
+            // Run the tests
+            nativeTest.testNativeExecution();
+        }
+        finally {
+            // Clean up resources
+            nativeTest.destroyTestFunctions();
+        }
+    }
+
+    private static class TestArraySortByKeyFunctionsNative
+            extends AbstractTestFunctions
+    {
+        public TestArraySortByKeyFunctionsNative()
+        {
+            // Enable native execution and explicitly disable the sidecar
+            super(testSessionBuilder()
+                    .setSystemProperty("native_execution_enabled", "true")
+                    .build(),
+                    new FeaturesConfig()
+                            .setNativeExecutionEnabled(true),
+                    new FunctionsConfig());
+        }
+
+        @Test
+        public void testNativeExecution()
```

Comment:
If these tests are just repeats of the test above, I think we can just remove them and remove this whole complicated nested test.  The flag actually doesn't even do anything for this test, you would need to actually spin up native workers as well, which I think is overcomplicated for this test.